src-built-in/components/UIAPI/hooks/workspaces.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from "react";
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState, WorkspaceState } from '../types';
+import * as Actions from '../actions/workspaceActions';
+
+
+/**
+ * A hook for getting the ActiveWorkspace,
+ * and setting the ActiveWorkspace.
+ */
+export const useWorkspaces = () => {
+	const dispatch = useDispatch();
+	const setActiveWorkspace = (activeWorkspace: WorkspaceState['activeWorkspace']) => {
+		dispatch(Actions.setActiveWorkspace(activeWorkspace));
+	}
+	// Run every time the workspace service pushes out an update.
+	const onWorkspaceUpdate = (err: any, response: any) => {
```

Comment:
Because the error isn't used here, it shouldn't be in the function. Also, we have to tighten up these types, but that can be done later.

Reply:
If there is an error, we should probably log it out to the console

Reply:
I'll let you guys deal with this later. err is the first argument always for a pub/sub even if you don't use it. Yeah we should log the error if there is one. I think tightening these types will have to wait for typing FSBL. Maybe not though